test/unit/range/views/view_kmer_hash_test.cpp

```diff
@@ -187,3 +192,12 @@ TEST_F(kmer_hash_test, issue1614)
     EXPECT_EQ(sequence | seqan3::views::kmer_hash(seqan3::ungapped{25}) | seqan3::views::to<std::vector<size_t>>,
               seqan3::views::repeat_n(298023223876953124, 26) | seqan3::views::to<std::vector<size_t>>);
 }
+
+// https://github.com/seqan/seqan3/issues/1643
+TEST_F(kmer_hash_test, issue1643)
+{
+    EXPECT_EQ(ungapped3, text3 | ungapped_view2 | seqan3::views::to<result_t>);
```

Comment:
It is really hard to understand what the test is about. So it becomes not clear that ungapped3 is empty and that the result should be that the resulting range is empty. Also it is not shown how the offending code from the referenced issue is fixed, so the reported use case should be used.
```cpp
    seqan3::dna4_vector text_23_elements{"ACGATCGATCGTAGCTACTGAGC"_dna4};
    
    auto k_mer_size_23_view = text_23_elements | seqan3::views::kmer_hash(seqan3::ungapped{23u};
    EXPECT_EQ(k_mer_size_23_view.size(), 1u);
    EXPECT_EQ(k_mer_size_23_view[0], 6829917194121u);

    auto k_mer_size_24_view = text_23_elements | seqan3::views::kmer_hash(seqan3::ungapped{24u};
    EXPECT_TRUE(k_mer_size_24_view.empty());

    auto k_mer_size_25_view = text_23_elements | seqan3::views::kmer_hash(seqan3::ungapped{25u};
    EXPECT_TRUE(k_mer_size_25_view.empty());
```